### ISSUE TITLE
Clarification on Handling Slices without Foreground Segmentation in Training Data

### ISSUE THREAD

**Issue by partha-ghosh** [microsoft/BiomedParse#63]
Hi,

I have a question regarding the training data requirements for BiomedParse. As described in the paper, the model is trained using image–mask–description triples, which suggests that each training image must have an associated segmentation mask. However, we are currently extracting image slices from 3D volumes along with their corresponding 3D segmentation volumes. As a result, most of the corresponding 2D image slices do not contain any foreground segmentation.

Could you please clarify the intended design? Specifically:

    Is the training dataset expected to include only images that contain foreground segmentations?

    In the case of slices with no foreground segmentation (i.e., no annotated objects), should these images be excluded from training, or is there a recommended way to incorporate them into the training process?

Thanks for your help in understanding the correct approach.

Best regards,
Partha

**Comment by theodore-zhao** [comment]
In our training we only used positive slices for segmentation purposes, where the prompted object does exist in the image. The way we used to filter out negative slices during inference is: let the model do segmentation inference, then compare the stats on the segmented region (pixel values, predicted foreground probability) with those from the training set.

For your purpose, you can either use only the positive slices, but that requires an extra step to determine if the segmented mask is valid or not. Alternatively you can also subsample some negative slices (not too much as I assume there are a lot more negative slices in a 3D volume). The corresponding mask will just be an all zero mask in that case. I think the current model is able to finetune on those type of data, but I am curious to know how well that works.

Hope this helps!
